spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,290 @@
+                            QWidget)
+
+# Local imports
+from spyder import __version__
+from spyder.api.translations import get_translation
+from spyder.utils.icon_manager import ima
+from spyder.utils.programs import is_module_installed
+
+logger = logging.getLogger(__name__)
+
+# Localization
+_ = get_translation('spyder')
+
+# Update installation process statuses
+NO_STATUS = __version__
+DOWNLOADING_INSTALLER = _("Downloading update")
+INSTALLING = _("Installing update")
+FINISHED = _("Installation finished")
+PENDING = _("Update available")
+CHECKING = _("Checking for updates")
+CANCELLED = _("Cancelled update")
+
+INSTALL_INFO_MESSAGES = {
+    DOWNLOADING_INSTALLER: _("Downloading Spyder {version}"),
+    INSTALLING: _("Installing Spyder {version}"),
+    FINISHED: _("Finished installing Spyder {version}"),
+    PENDING: _("Spyder {version} available to download"),
+    CHECKING: _("Checking for new Spyder version"),
+    CANCELLED: _("Spyder update cancelled")
+}
+
+
+class UpdateInstallationCancelledException(Exception):
+    """Update installation was cancelled."""
+    pass
+
+
+class UpdateInstallation(QWidget):
+    """Update progress installation widget."""
+
+    def __init__(self, parent):
+        super().__init__(parent)
+
+        # Left side
+        action_layout = QVBoxLayout()
+        progress_layout = QHBoxLayout()
+        self._progress_widget = QWidget(self)
+        self._progress_widget.setFixedHeight(50)
+        self._progress_bar = QProgressBar(self)
+        self._progress_bar.setFixedWidth(180)
+        self.cancel_button = QPushButton()
+        self.cancel_button.setIcon(ima.icon('DialogCloseButton'))
+        progress_layout.addWidget(self._progress_bar, alignment=Qt.AlignLeft)
+        progress_layout.addWidget(self.cancel_button)
+        self._progress_widget.setLayout(progress_layout)
+
+        self._progress_label = QLabel(_('Downloading'))
+
+        self.install_info = QLabel(
+            _("Downloading Spyder update <br>"))
+
+        button_layout = QHBoxLayout()
+        self.ok_button = QPushButton(_('OK'))
+        button_layout.addStretch()
+        button_layout.addWidget(self.ok_button)
+        button_layout.addStretch()
+        action_layout.addStretch()
+        action_layout.addWidget(self._progress_label)
+        action_layout.addWidget(self._progress_widget)
+        action_layout.addWidget(self.install_info)
+        action_layout.addSpacing(10)
+        action_layout.addLayout(button_layout)
+        action_layout.addStretch()
+
+        # Layout
+        general_layout = QHBoxLayout()
+        general_layout.addLayout(action_layout)
+
+        self.setLayout(general_layout)
+
+    def update_installation_status(self, status, latestVersion):
+        """Update installation status (downloading, installing, finished)."""
+        self._progress_label.setText(status)
+        self.install_info.setText(INSTALL_INFO_MESSAGES[status].format(
+            version=latestVersion))
```

Comment:
Lets change the variable name here to `latest_version` instead of `latestVersion`. So:
```suggestion
    def update_installation_status(self, status, latest_version):
        """Update installation status (downloading, installing, finished)."""
        self._progress_label.setText(status)
        self.install_info.setText(INSTALL_INFO_MESSAGES[status].format(
            version=latest_version))
```